feat(*): add k8s version to version debug cmd

rbwsam opened the pull request:
When 'helm version --debug' is run, this will print the kubernetes server
version as well as the helm client and server versions.

Closes #1398 

---

I'm wide open to feedback - I'm here to learn!

I think it would make sense to also move the tiller version code to the new `InfoService` but I didn't want to include that in this PR since it's probably a breaking change.

rbwsam committed:
feat(*): add k8s version to version debug cmd

When 'helm version --debug' is run, this will print the kubernetes server
version as well as the helm client and server versions.

Closes #1398

rbwsam commented on pkg/proto/hapi/chart/chart.pb.go:
```diff
@@ -1,5 +1,6 @@
-// Code generated by protoc-gen-go. DO NOT EDIT.
```
I made sure I built this with libprotoc 3.2.0 but the comments are still different...

technosophos commented:
Sorry for being late on this one. Would it be easier to just run the query from the client directly to the Kubernetes API server instead of adding a Tiller roundtrip? (For example, see the tunnel code). Then we wouldn't have to modify the proto at all.

rbwsam commented:
Great point! I will make that change and update the PR.

rbwsam committed:
Revert "feat(*): add k8s version to version debug cmd"

This reverts commit 63be90dda50885ccef7c2dee08c34a6b8e82b3a6.

rbwsam committed:
feat(*): add k8s version to version debug cmd

When 'helm version --debug' is run, this will print the kubernetes server
version as well as the helm client and server versions.

Closes #1398

bacongobbler commented on cmd/helm/version.go:
```diff
@@ -89,6 +90,13 @@ func newVersionCmd(c helm.Interface, out io.Writer) *cobra.Command {
 }
 
 func (v *versionCmd) run() error {
+	if settings.Debug {
```
hey! Sorry about the long wait on the review here.

Would you mind moving this after line 106? If we're making a server-side call to check the kubernetes version, we will want to respect the `--client-only` flag.

jascott1 commented:
Hi @rbwsam Thanks for the PR! It looks like we are one small change away from being mergeable. Do you think you will be able to continue working on this?

rbwsam commented:
@jascott1 Yes I am happy to make the changes, expect them in the next 48 hours.

Commit:
Only show k8s version if asking for server version

bacongobbler reviewed (approved):
This is a fantastic enhancement. Thanks @rbwsam!
